src/embeddings/embeddings_calculator.cc

```diff
@@ -123,42 +128,76 @@ class EmbeddingsCalculator : public CalculatorBase {
+                RET_CHECK(tokenizerOutputMap.size() >= embeddingsInputNames.size());
+                RET_CHECK(tokenizerOutputMap[EMBEDDINGS_MODEL_INPUT_IDS_NAME].get_shape().size() == 2);
+                size_t input_ids_size = tokenizerOutputMap[EMBEDDINGS_MODEL_INPUT_IDS_NAME].get_shape()[1];
+                if (input_ids_size > max_context_length) {
+                    SPDLOG_LOGGER_DEBUG(embeddings_calculator_logger, "Input size {} exceeds max_context_length {}", input_ids_size, max_context_length);
+                    return absl::InvalidArgumentError(absl::StrCat("Input length ", input_ids_size, " longer than allowed ", max_context_length));
+                }
+                for (const auto& embeddingsInputName : embeddingsInputNames) {
+                    auto it = tokenizerOutputMap.find(embeddingsInputName);
+                    RET_CHECK(it != tokenizerOutputMap.end());
+                    SPDLOG_LOGGER_DEBUG(embeddings_calculator_logger, "Embedding model input {} is connected with matching tokenizer output", embeddingsInputName);
+                    embeddingsInputMap[embeddingsInputName] = it->second;
+                    if (embeddingsInputName == "attention_mask") {
+                        if (received_batch_size == 1) {
+                            handler.setPromptTokensUsage(it->second.get_size());
+                            continue;
                         }
-                    } else {
-                        for (int i = 0; i < it->second.get_byte_size(); i++) {
-                            attendedTokens += reinterpret_cast<uint8_t*>(it->second.data())[i];
+                        size_t attendedTokens = 0;
+                        if (it->second.get_element_type() == ov::element::Type_t::i64) {
+                            for (int i = 0; i < it->second.get_size(); i++) {
+                                attendedTokens += reinterpret_cast<int64_t*>(it->second.data())[i];
+                            }
+                        } else if (it->second.get_element_type() == ov::element::Type_t::i32) {
+                            for (int i = 0; i < it->second.get_size(); i++) {
+                                attendedTokens += reinterpret_cast<int32_t*>(it->second.data())[i];
+                            }
+                        } else {
+                            for (int i = 0; i < it->second.get_byte_size(); i++) {
+                                attendedTokens += reinterpret_cast<uint8_t*>(it->second.data())[i];
+                            }
                         }
+                        handler.setPromptTokensUsage(attendedTokens);
                     }
-                    handler.setPromptTokensUsage(attendedTokens);
+                }
+            } else if (auto ints = std::get_if<std::vector<std::vector<int64_t>>>(&input)) {
+                size_t token_count_of_longest_document = 0;
+                int64_t pad_token = modelConfig["pad_token_id"].as<int64_t>();
```

Comment:
What is the type of modelConfig? Is it ov::AnyMap? Is it std::map?

- in a case when it is ov::AnyMap - what does operator[] do? does it throw exception when key does not exist? does it insert with default value? what is the default value in case it inserts it? is it desired default value? do we have unit test for that behavior?

- in case it is std::map - the same questions as above

Reply:
So would be the case for this one https://github.com/openvinotoolkit/model_server/blob/main/src/embeddings/embeddings_calculator.cc#L164?

Reply:
In line 164 you mentioned, we first check if the key exists using `modelConfig.count("max_position_embeddings")`